Nonlinear least squares classes (#677)
fitbenchmarking/fitbenchmarking: docs/source/users/options/fitting_option.rst

This sets the cost function to be used for the given data. Current cost
functions supported are:

* ``nlls`` - This sets the cost function to be non-weighted non-linear least squares, :class:`~fitbenchmarking.cost_func.nlls_cost_func.NLLSCostFunc`.

> * ``weigthed_nlls`` - This sets the cost function to be weighted non-linear least squares, :class:`~fitbenchmarking.cost_func.weighted_nlls_cost_func.WeightedNLLSCostFunc`.

Reviewer: ```suggestion
* ``weighted_nlls`` - This sets the cost function to be weighted non-linear least squares, :class:`~fitbenchmarking.cost_func.weighted_nlls_cost_func.WeightedNLLSCostFunc`.
```
```suggestion
* ``weigthed_nlls`` - This sets the cost function to be weighted non-linear least squares, :class:`~fitbenchmarking.cost_func.weighted_nlls_cost_func.WeightedNLLSCostFunc`.
```

> * ``root_nlls`` - This sets the cost function to be root non-linear least squares, :class:`~fitbenchmarking.cost_func.root_nlls_cost_func.RootNLLSCostFunc`.

Reviewer: ```suggestion
* ``hellinger_nlls`` - This sets the cost function to be the Hellinger cost function, :class:`~fitbenchmarking.cost_func.root_nlls_cost_func.HellingerNLLSCostFunc`.
```

Default is ``nlls``

.. code-block:: rst

    [FITTING]

>     cost_func_type: root_nlls

Reviewer: ```suggestion
    cost_func_type: weighted_nlls
```
